### PR TITLE
PageObject of Register page

### PR DESCRIPTION
**Pull request by ritaadhikari**
### Description
Here I have created PageObject for registrationPage and also created pageobjectof homepage  for logout button

### CODE REVIEW

File: Features2/tests/features/userRegistration.feature [hunk 1]
```diff
@@ -22,15 +22,18 @@ Feature:User registration
...
-            When the user enters with a "<email>"
-            Then the user will be given a "<message>"
-            Examples:
-                | email                   | message|
-                | abchefg98gmailcom       | Please enter an email address |
-                | rits98.com              | Please enter an email address |
-                | rira@gmail              |                               |
-                | risahgdgajda36gmail.com | Please enter an email address |
+
+# this was commented because we haven't created stepDefinations for this scenario
```
Comment (SagarGi) [inline Features2/tests/features/userRegistration.feature]:
```suggestion
# This is a valid scenario (test) but the selector of input filed pop up could not be extracted which is why this is commented.
```

File: Features2/tests/pageObject/RegisterPage.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    this.messageSelector = 'div[class="alert alert-danger"]';
+    this.registerUrl = "http://localhost:3000/register";
+  }
+  async clickRegisterButton() {
+    await page.click(this.registerSelector);
+  }
+
+  async navigateToRegisterPage() {
+    await page.goto(this.registerUrl);
+  }
```
Comment (SagarGi) [inline Features2/tests/pageObject/RegisterPage.js]: this two function basically does the same may be use only one

File: Features2/tests/stepDefinations/loginContext.js [hunk 3]
```diff
@@ -1,8 +1,10 @@
 const { Given, When, Then } = require("@cucumber/cucumber");
 const { expect } = require("@playwright/test");
 const { LoginPage } = require("../pageObject/LoginPage");
+const { HomePage } = require("../pageObject/HomePage");
 
 const loginPage = new LoginPage();
+const homePage = new HomePage();
 
 Given("a user has navigated to the login dashboard", async function () {
```
Comment (SagarGi) [inline Features2/tests/stepDefinations/loginContext.js:9]: if this step is unused remove it

File: Features2/tests/stepDefinations/registerContext.js [hunk 4]
```diff
@@ -1,57 +1,48 @@
...
-const passwordSelector = 'input[name="password"]';
-const confirmSelector = 'input[name="password2"]';
-const logoutSelector = 'span[class="hide-sm"]';
-const registerButtomSelector = 'input[value="Register"]';
-const messageSelector = 'div[class="alert alert-danger"]';
 
 Given("the user has clicked Register button", async function () {
-  await page.click(registerSelector);
+  await registerPage.clickRegisterButton();
 });
```
Comment (SagarGi) [inline Features2/tests/stepDefinations/registerContext.js]: do not make step for every actions

File: Features2/tests/stepDefinations/loginContext.js (hunk 3 above)
Comment (ritaadhikari) [reply Features2/tests/stepDefinations/loginContext.js:9]: it is used

**Review by SagarGi — APPROVED** [review]

**Pull request merged**